packages/tools/devtools/devtools-core/src/data-visualization/DataVisualization.ts

```diff
@@ -482,8 +511,24 @@ export async function visualizeChildData(
  * Determines whether or not the provided value is an {@link ISharedObject}, for the purposes of this library.
  * @remarks Implemented by checking for the particular properties / methods we use in this module.
  */
-function isSharedObject(value: Partial<ISharedObject>): value is ISharedObject {
+function isSharedObject(value: unknown): value is ISharedObject {
+	return (
+		(value as SharedObject).id !== undefined &&
+		(value as SharedObject).attributes?.type !== undefined &&
+		(value as SharedObject).on !== undefined
+	);
+}
+
+/**
+ * Determines whether or not the provided value is an {@link DataObject}, for the purposes of this library.
+ * @remarks Implemented by checking for the particular properties / methods we use in this module
+ */
+function isDataObject(value: unknown): value is DataObject {
```

Comment:
Since `DataObject` is a class, we should actually be able to replace the use of this function with `value instanceof DataObject`.

Reply:
Correct me if I'm wrong here, but I thought we should generally avoid using `instanceof DataObject` for typeguard (if there is a version mismatch between the Fluid application and Devtools package, each can refer to different instance of the `DataObject` although the class definitions are identical).

Reply:
Yeah, I don't think checking for the class is a good idea here.

Reply:
It is generally tue that `instanceof` is fragile in certain ways. Namely, if devtools somehow depended on a different version of `aqueduct` than the end-user, then that check wouldn't work correctly. But we don't generally allow mixing and matching of our libraries that way anyways. And worst case, incompatible data object implementations simply wouldn't get a visualizer rendered.

I don't feel strongly that we _have_ to simplify this, but I think it would be safe enough. Either way :)

Reply:
Oh and this is in core, not in something related to the extension... yeah, now that I realize that I don't feel as strongly about not using `instanceof`. I think I'd lean towards not using it if possible anyway, because our guidance may say we don't support mixed versions, but eventually someone will miss it and not understand why things aren't working. Preventing that altogether seems worth it to me.